src/app/meshes/components/MeshDetails.vue

```diff
@@ -127,53 +185,26 @@ const props = defineProps({
   },
 })
 
-const basicMesh = computed(() => {
-  const { name, creationTime, modificationTime } = props.mesh
-  return {
-    name,
-    created: formatIsoDate(creationTime),
-    modified: formatIsoDate(modificationTime),
-    'Data Plane Proxies': props.meshInsight.dataplanes.total,
-  }
-})
-
-const hasLocalityAwareLoadBalancing = computed(() => Boolean(props.mesh.routing?.localityAwareLoadBalancing))
-
-const extendedMesh = computed(() => ({
-  mtls: getBackendData(props.mesh, 'mtls'),
-  logging: getBackendData(props.mesh, 'logging'),
-  metrics: getBackendData(props.mesh, 'metrics'),
-  tracing: getBackendData(props.mesh, 'tracing'),
-}))
+const mtls = computed(() => getBackendData(props.mesh.mtls))
+const logging = computed(() => getBackendData(props.mesh.logging))
+const metrics = computed(() => getBackendData(props.mesh.metrics))
+const tracing = computed(() => getBackendData(props.mesh.tracing))
 
 const totalPolicyCount = computed(() => {
   return Object.values(props.meshInsight.policies ?? {}).reduce((total, stat) => total + stat.total, 0)
 })
-const policyTypes = computed(() => {
-  return Object.entries(props.meshInsight.policies ?? {})
-    .map(([policyTypeName, stat]) => {
-      const policyType = store.state.policyTypesByName[policyTypeName]
-
-      if (policyType && stat.total !== 0) {
-        return {
-          name: policyType.name,
-          path: policyType.path,
-          total: stat.total,
-        }
-      }
-
-      return null
-    })
-    .filter(notEmpty)
-})
 
-function getBackendData(mesh: Mesh, field: 'mtls' | 'logging' | 'metrics' | 'tracing') {
-  if (mesh[field] === undefined) {
+function getBackendData(meshBackend?: MeshBackend) {
+  if (!meshBackend || !Array.isArray(meshBackend.backends) || meshBackend.backends.length === 0) {
```

Comment:
nit: we could extract it to `meshBackendEmptyOrNull` just like couple of lines below for enabledBackendName.

Reply:
If it’s not a big improvement in readability I prefer to leave this in the if statement because the evaluation can then short-circuit on the first truthy result. Depending on how this code will be minified (i.e. will the expression be inlined because it’s clearly only in function scope and thus safe to do?), this won’t be possible.